napari/_qt/layer_controls/qt_labels_controls.py

```diff
@@ -23,7 +24,16 @@
 from ..widgets.qt_mode_buttons import QtModePushButton, QtModeRadioButton
 from .qt_layer_controls_base import QtLayerControls
 
-INT32_MAX = 2 ** 31 - 1
+
+def double_as_uint_spinbox(dtype: np.dtype):
+    iinfo = np.iinfo(dtype)
+
+    spinbox = QDoubleSpinBox()
+    spinbox.setSingleStep(1)
+    spinbox.setMinimum(max(iinfo.min, 0))
+    spinbox.setMaximum(min(iinfo.max, MAX_FLOAT64_INT))
```

Comment:
unless we use it anywhere else, let's just move `MAX_FLOAT64_INT` to this module (could be a class attribute too)